Source/JavaScriptCore/runtime/JSDateMath.cpp

```diff
@@ -392,7 +411,17 @@ void DateCache::msToGregorianDateTime(double millisecondsFromEpoch, WTF::TimeTyp
         localTime = localTimeOffset(static_cast<int64_t>(millisecondsFromEpoch));
         millisecondsFromEpoch += localTime.offset;
     }
-    tm = GregorianDateTime(millisecondsFromEpoch, localTime);
+    if (std::isfinite(millisecondsFromEpoch)) {
+        WTF::Int64Milliseconds timeClipped(static_cast<int64_t>(millisecondsFromEpoch));
+        int32_t days = WTF::msToDays(timeClipped);
+        int32_t timeInDayMS = WTF::timeInDay(timeClipped, days);
+        auto [year, month, day] = yearMonthDayFromDaysWithCache(days);
+        int32_t hour = timeInDayMS / (60 * 60 * 1000);
+        int32_t minute = (timeInDayMS / (60 * 1000)) % 60;
+        int32_t second = (timeInDayMS / 1000) % 60;
+        tm = GregorianDateTime(year, month, dayInYear(year, month, day), day, WTF::weekDay(days), hour, minute, second, localTime.offset / WTF::Int64Milliseconds::msPerMinute, localTime.isDST);
```

Comment:
In the next patch, we will introduce JSC tailored PlainGregorianDateTime, and potentially purge RefPtr<DateInstanceData> completely (And make DateInstance non-destructible).

Reply:
Can we share this helper code?

Reply:
Probably, we will have different implementation from GregorianDateTime soon, so they will not be duplicate.